Files linux-2.6.12.4-vs2.0/include/linux/byteorder/.generic.h.swp and linux-2.6.12.4-vs2.0.0.2/include/linux/byteorder/.generic.h.swp differ
diff -NurpP --minimal linux-2.6.12.4-vs2.0/include/net/route.h linux-2.6.12.4-vs2.0.0.2/include/net/route.h
--- linux-2.6.12.4-vs2.0/include/net/route.h	2005-08-07 20:06:39 +0200
+++ linux-2.6.12.4-vs2.0.0.2/include/net/route.h	2005-08-18 16:11:26 +0200
@@ -146,7 +146,7 @@ static inline char rt_tos2priority(u8 to
 	return ip_tos2prio[IPTOS_TOS(tos)>>1];
 }
 
-#define IPI_LOOPBACK	0x0100007f
+#define IPI_LOOPBACK	htonl(INADDR_LOOPBACK)
 
 static inline int ip_find_src(struct nx_info *nxi, struct rtable **rp, struct flowi *fl)
 {
diff -NurpP --minimal linux-2.6.12.4-vs2.0/net/ipv4/af_inet.c linux-2.6.12.4-vs2.0.0.2/net/ipv4/af_inet.c
--- linux-2.6.12.4-vs2.0/net/ipv4/af_inet.c	2005-08-07 20:06:39 +0200
+++ linux-2.6.12.4-vs2.0.0.2/net/ipv4/af_inet.c	2005-08-18 16:24:38 +0200
@@ -425,7 +425,7 @@ int inet_bind(struct socket *sock, struc
 			s_addr = ipv4root;
 			s_addr1 = (nbipv4 > 1) ? 0 : s_addr;
 			s_addr2 = v4_bcast;
-		} else if (s_addr == 0x0100007f) {
+		} else if (s_addr == IPI_LOOPBACK) {
 			/* rewrite localhost to ipv4root */
 			s_addr = ipv4root;
 			s_addr1 = ipv4root;
